pkg/scheduler/api/types.go

```diff
@@ -53,10 +59,40 @@ const (
 	Unknown
 )
 
+// JobReadiness type of job readiness
+type JobReadiness int
+
+const (
+	// Ready : a job is Ready if the number of tasks in Allocated state
+	// exceeds the job's minimum task number requirement. In other words,
+	// #(Allocated Tasks) >= Job.MinAvailable
+	// A Ready job can be dispatch to a node right away.
+	Ready JobReadiness = 1 << iota
+
+	// OverResourceReady : a job is OverResourceReady if the job is not Ready for dispatch, but
+	// the number of tasks in Allocated state exceeds the job's minim task
+	// number requirement. In other words,
+	// #(Allocated Tasks) < Job.MinAvailable &&
+	// #(Allocated Tasks) + #(AllocatedOverBackFill Tasks) >= Job.MinAvailable
+	OverResourceReady
```

Comment:
I can live with this name. But personally I still prefer `ConditionallyReady`. :-)